hudi-common/src/main/java/org/apache/hudi/io/storage/HoodieNativeAvroHFileReader.java

```diff
@@ -190,18 +193,19 @@ public Schema getSchema() {
 
   @Override
   public void close() {
-    try {
-      if (sharedHFileReader.isPresent()) {
-        sharedHFileReader.get().close();
-      }
-    } catch (IOException e) {
-      throw new HoodieIOException("Error closing the HFile reader", e);
-    }
+    metaInfoMap.clear();
```

Comment:
How many keys do we expect in the map. Clear nulls out all the entries but doesn't reduce size. That might be fine if it's only in the thousands. But if it's going to be a lot then maybe we make this not final and do metaInfoMap = null so it can be GCd

Reply:
Oh nevermind PRELOADED_META_INFO_KEYS is the size of the map.